oteljava/trace/src/main/scala/org/typelevel/otel4s/oteljava/trace/SpanBuilderImpl.scala

```diff
@@ -59,7 +59,7 @@ private[oteljava] final case class SpanBuilderImpl[F[_]: Sync] private (
       Resource.eval(runnerContext).flatMap(ctx => runner.start(ctx))
 
     override def use[A](f: Span[F] => F[A]): F[A] =
-      resource.use { res => res.trace(f(res.span)) }
+      resource.use(res => res.trace(Sync[F].defer(f(res.span))))
```

Comment:
Otherwise, the context isn't captured. 
Without this change, the following example doesn't work:
```scala
tracer.span("test").use { span => // start 'test' span using otel4s
  val jSpanContext = JSpan.current().getSpanContext // get a span from a ThreadLocal var
  IO.println(s"jCtx: ${jSpanContext}, Otel4s ctx: ${span.context}")
}
```
`jCtx` will be 'invalid' in the case.

Without the `defer`, a function `f` is materialized before the `res.trace` is applied. `res.trace` sets the correct context. Since the function is evaluated before the context is set, it captures the wrong (old) context.